### PR TITLE
[fix] Show proper error if component has no class and no creator

### PR DESCRIPTION
**Pull request by pieleric**
If a component is created by another component, but in the microscope
file it's not declared in the "children", then the backend doesn't know
which component is its creator.

Such error used to be handled properly before because most components
always has as "parent" also the Microscope which *weirdly* declared
every component as its child. However, now that we have removed this
weirdness, a component without creator would have no key "parents". This
raised an exception earlier than expected.

=> handle such case by returning an empty parents list. Then the usual
error is displayed properly again.

### CODE REVIEW

File: src/odemis/odemisd/modelgen.py [hunk 1]
```diff
@@ -376,14 +376,14 @@ def _fill_creator(self):
                 parents = [p for p in parents if self.ast[p].get("class") != "Microscope"]
 
                 if len(parents) == 0:
-                    raise SemanticError("Error in microscope file: component %s "
-                            "has no class specified and is not created by any "
-                            "component." % name)
+                    raise SemanticError("Error in microscope file: component \"%s\" "
+                            "has no class specified and not component declares in its "
+                            "children." % name)
```
Comment (sneepstefan) [inline src/odemis/odemisd/modelgen.py:381]:
```suggestion
                    raise SemanticError("Error in microscope file: component \"%s\" "
                            "has no class specified and not component declares in its "
                            "children." % name)
```
This sentence looks a little weird to me, could you rephrase that?

Comment (pieleric) [reply src/odemis/odemisd/modelgen.py:381]: How about: `"Error in microscope file: component \"%s\ has no class specified and is not child of any component."` ?

Comment (sneepstefan) [reply src/odemis/odemisd/modelgen.py:381]: yeah that sounds much clearer :)
